src/app/colors/mpl-colormaps/mpl-colormap.model.ts

```diff
@@ -0,0 +1,179 @@
+                );
+        }
+    }
+
+    private static mplColorToRgb(mplColor: [number, number, number]): [number, number, number] {
+        return [mplColor[0] * 255, mplColor[1] * 255, mplColor[2] * 255];
+    }
+
+    static createColorizerDataWithName(
+        mplColormapName: MplColormapName,
+        min: number, max: number,
+        steps: number | undefined = 16,
+        stepScale: ColormapStepScale = 'log_values'
+    ) {
+
+        const colormap = MplColormap.getMplColormapForName(mplColormapName);
+        const trueSteps = (steps && steps <= colormap.length) ? steps : colormap.length;
+        const stepFractions = MplColormap.calculateStepScales(trueSteps, min, max, stepScale);
+
+        return MplColormap.createColorizerDataFromColormapAndStepFractions(colormap, min, max, stepFractions);
+    }
+
+    private static generateLinearStepFractions(steps: number): Array<number> {
+        const maxIndex = steps - 1;
+        const stepFractions = new Array<number>(steps);
+        stepFractions[0] = 0;
+        stepFractions[maxIndex] = 1;
+
+        for (let i = 1; i < maxIndex; i++) { // fill the values between 0 and 1.
+            stepFractions[i] = i / (maxIndex);
+        }
+
+        return stepFractions;
+    }
+
+    private static generatePowerStepFractionsSteps(steps: number, pow: number = 2): Array<number> {
+        const maxIndex = steps - 1;
+        const stepFractions = new Array<number>(steps);
+        stepFractions[0] = 0;
+        stepFractions[maxIndex] = 1;
+
+        for (let i = 0; i < maxIndex - 1; i++) {
+            stepFractions[i + 1] = Math.pow(i, pow) / (Math.pow(maxIndex, pow));
+        }
+
+        return stepFractions;
+    }
+
+    private static generatePowStepFractionsValues(steps: number, min: number, max: number, pow: number = 2): Array<number> {
+        const stepFractions = new Array<number>(steps);
+        const valueRange = max - min;
+        const rangeStep = valueRange / (steps - 1);
+
+        for (let i = 0; i < steps; i++) {
+            const currentVal = i * rangeStep;
+            const temp1 = (currentVal < 0) ? 0 : currentVal;
+            const temp2 = Math.pow(temp1, pow);
+            stepFractions[i] = temp2 / Math.pow(valueRange, pow);
+        }
+        return stepFractions;
+    }
+
+    private static inverseStepFractions(stepFractions: Array<number>): Array<number> {
+        const inverseFraction = stepFractions.map(x => 1.0 - x).reverse();
+        return inverseFraction;
+    }
+
+    private static generateLogStepFractionsValues(steps: number, min: number, max: number): Array<number> {
+
+        const valueRange = max - min;
+        const rangeStep = valueRange / (steps - 1);
+
+        const stepFractions = new Array<number>(steps);
+        for (let i = 0; i < steps; i++) {
+            const currentVal = i * rangeStep + min;
+            stepFractions[i] = (Math.log(currentVal) - Math.log(min)) / (Math.log(max) - (Math.log(min)));
```

Comment:
Macht es einen Unterschied, ob wir wie hier Log zur Basis e nehmen oder Log10?

Reply:
für mich nicht. MPL LogNorm macht aber Log zur Basis e.

Reply:
Siehe oben, in der Doku steht, dass es zur Basis 10 rechnet.
>  Using colors.LogNorm() normalizes the data via log10. 

https://matplotlib.org/3.1.0/tutorials/colors/colormapnorms.html

Reply:
sollte sich erledigt haben